src/features/recovery/RecoveryDialogs.tsx

```diff
@@ -0,0 +1,350 @@
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    addConfirmBeforeClose,
+    Alert,
+    Button,
+    clearConfirmBeforeClose,
+    ConfirmationDialog,
+    DialogButton,
+    GenericDialog,
+    logger,
+    useStopwatch,
+} from '@nordicsemiconductor/pc-nrfconnect-shared';
+
+import {
+    isSessionRecoveryPending,
+    setSavePending,
+} from '../../slices/appSlice';
+import { formatDuration, formatTimestamp } from '../../utils/formatters';
+import TimeComponent from '../ProgressDialog/TimeComponent';
+import { RecoveryManager } from './RecoveryManager';
+import {
+    DeleteAllSessions,
+    RemoveSessionByFilePath,
+    Session,
+} from './SessionsListFileHandler';
+
+const SessionItem = ({
+    session,
+    onRecoverClick,
+    onRemoveClick,
+}: {
+    session: Session;
+    onRecoverClick: (session: Session) => void;
+    onRemoveClick: (session: Session) => void;
+}) => (
+    <div className="tw-flex tw-flex-col tw-justify-end tw-gap-11 tw-bg-gray-800 tw-p-3 tw-text-white">
+        <div className="tw-flex tw-flex-grow tw-flex-col tw-justify-between tw-gap-2">
+            <div className="tw-flex tw-flex-row tw-justify-between">
+                <div>
+                    <div className="tw-text-xs">Start time</div>
+                    <div>{formatTimestamp(session.startTime)}</div>
+                </div>
+                <div>
+                    <div className="tw-text-xs">Duration</div>
+                    <div>{formatDuration(session.samplingDuration || 0)}</div>
+                </div>
+                <div>
+                    <div className="tw-text-xs">Sampling rate</div>
+                    <div>{session.samplingRate}</div>
+                </div>
+                <div className="tw-content-center tw-align-middle">
+                    <div className="tw-flex tw-flex-row tw-gap-2">
+                        <Button
+                            variant="secondary"
+                            onClick={() => onRemoveClick(session)}
+                        >
+                            Delete
+                        </Button>
+                        <Button
+                            variant="primary"
+                            className="tw-w-[60px]"
+                            onClick={() => onRecoverClick(session)}
+                        >
+                            {session.alreadyRecovered ? 'Load' : 'Recover'}
+                        </Button>
+                    </div>
+                </div>
+            </div>
+            {session.alreadyRecovered && (
+                <div className="tw-flex tw-flex-row tw-gap-1 tw-text-orange-400">
+                    <span className="mdi mdi-information-outline info-icon" />
+                    <span className="tw-grid tw-items-center tw-text-xs">
+                        The session has already been recovered or loaded by
+                        using a ppk2 file. You can load it again or delete it.
```

Comment:
```suggestion
                        The session has already been recovered or loaded by
                        using a PPK2 file. You can load it again or delete it.
```
I don't think this notification is needed. I think we're going into too much use-case hell here. If the session has been already recovered, why not give the user the option to save here already? Why drag him into the app again. We can go into so many specifics here, but the bottom line is: a session has closed, we recovered it, you can not click Recover to have it in the app.